### PR TITLE
Optimizations and logs

### PR DESCRIPTION
**Pull request by anonthing**

### CODE REVIEW

File: myst/src/query/query.rs [hunk 1]
```diff
@@ -133,7 +133,7 @@ impl Query {
         config: &Config,
         metrics_reporter: Option<&Box<dyn MetricsReporter>>,
     ) -> Result<
-        tokio::sync::mpsc::Receiver<
```
Comment (gskchaitanya) [inline myst/src/query/query.rs:136]: Any implications of having this Unbounded? large enough response can fill up the memory?

**Review by gskchaitanya — APPROVED** [review]

File: myst/src/query/query.rs (hunk 1 above)
Comment (anonthing) [reply myst/src/query/query.rs:136]: that is possible. I'll add checks later to have some sort of restriction. The only caveat with using a bounded channel is the server threads will be blocked on `send` call if the clients are not polling the response fast enough.

Comment (mailravi3390) [reply myst/src/query/query.rs:136]: If you are using gRPC compression, you may not need to do anything if gRPC is doing streaming compression (which it should be doing).  Though it maybe worthwhile to figure out at what level compression is done. i.e at individual message level or across multiple messages. If it is across multiple messages, then the queue will get filled up depending on the amount of time compression takes.

Comment (anonthing) [reply myst/src/query/query.rs:136]: Not using compression. Our tests showed that compression has an impact on the latency, making the queries 10x slower in some cases. Tonic only supports gzip currently, there is some work on snappy for next release. AFAIK, compression is done on a stream level.

**Pull request merged**